frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -1183,6 +1311,13 @@ impl<T: Config> Pallet<T> {
 	}
 
 	fn do_elect() -> Result<(Supports<T::AccountId>, Weight), ElectionError> {
+		// We have to unconditionally try finalizing the signed phase here. There are only two
+		// possibilities:
+		//
+		// - signed phase was open, in which case this is essential for correct functioning of the system
+		// - signed phase was complete or not started, in which case finalization is idempotent and
+		//   inexpensive (1 read of an empty vector).
+		let (_, signed_finalize_weight) = Self::finalize_signed_phase();
```

Comment:
We can probably factor this out and if `Phase::Emergency`, don't even call to this. But my point is that, I _think_, it is guaranteed that if `Phase::Emergency` then the queue is empty and it will remain empty.

Reply:
I agree: in an emergency phase, the queue should be empty, making everything cheap enough to ignore.